core/src/main/java/com/github/jsonldjava/utils/JsonUtils.java

```diff
@@ -371,6 +363,56 @@ public static Object fromURL(java.net.URL url, CloseableHttpClient httpClient)
         }
     }
 
+    private static InputStream getJsonLdViaHttpUri(final URL url, final CloseableHttpClient httpClient,
+            CloseableHttpResponse response) throws IOException {
+        final HttpUriRequest request = new HttpGet(url.toExternalForm());
+        // We prefer application/ld+json, but fallback to application/json
+        // or whatever is available
+        request.addHeader("Accept", ACCEPT_HEADER);
+        response = httpClient.execute(request);
+
+        final int status = response.getStatusLine().getStatusCode();
+        if (status != 200 && status != 203) {
+            throw new IOException("Can't retrieve " + url + ", status code: " + status);
+        }
+        // follow alternate document location
+        // https://www.w3.org/TR/json-ld11/#alternate-document-location
+        URL alternateLink = alternateLink(url, response);
+        if (alternateLink != null) {
+            return getJsonLdViaHttpUri(alternateLink, httpClient, response);
+        }
+        return response.getEntity().getContent();
+    }
+
+    private static URL alternateLink(URL url, CloseableHttpResponse response)
+            throws MalformedURLException, IOException {
+        if (response.getEntity().getContentLength() > 0
+                && !response.getEntity().getContentType().getValue().equals("application/ld+json")) {
```

Comment:
Why does there need to be a content entity attached to this response and it is already a JSON-LD document if we are looking through the headers for a JSON-LD link?

Reply:
Ithink by this

> Documents which can't be directly interpreted as JSON-LD can provide an alternate location containing JSON-LD

(https://www.w3.org/TR/json-ld11/#alternate-document-location)

it is meant to not follow any link headers (even if, wrongly, present).

Reply:
My bad. I misread the if condition here. You are correct.